text/0000-project-ffi-unwind.md

```diff
@@ -0,0 +1,211 @@
+well-defined by causing Rust functions defined with `extern "C"` to abort the
+application whenever an uncaught `panic` would otherwise escape into the
+caller. Unfortunately, previous attempts to stabilize this behavior have caused
+existing, working projects to break.
+
+The problem here is not that the existing projects break *per se*: they are
+relying on [Undefined Behavior], so breakage is to be expected as a
+possibility. The problem is that there is no alternative available to them that
+would allow them to keep working (even if they are continuing to rely on
+behavior that is not yet fully specified).
+
+Previous attempts to provide a well-defined mechanism for unwinding across FFI
+boundaries have failed to reach consensus. Notably, two proposed RFCs generated
+over 400 comments between them before ultimately being closed:
+
+* [rust-lang/rfcs#2699](https://github.com/rust-lang/rfcs/pull/2699)
+* [rust-lang/rfcs#2753](https://github.com/rust-lang/rfcs/pull/2753)
+
+GitHub comment threads become difficult to follow for discussions this lengthy,
+and the disagreements in these threads have felt less productive than we
+believe they could be if more structure were provided.
+
+We would also like to demonstrate the Rust lang team's commitment to providing
+such a mechanism without needing to agree in advance on what language changes
+will be stabilized in order to do so.
+
+# Prototyping 'shepherded' project groups
+[prototyping-project-groups]: #prototyping-shepherded-project-groups
+
+With this RFC, we formally announce the formation of a project-specific,
+shepherded "project group" to adopt responsibility for driving progress on
+specifying unwinding behavior at FFI boundaries.
+
+## What is a "project group"?
+
+The "project group" term has not previously been used: it is intended to
+formalize a concept that has existed informally for some time, under a number
+of names (including "working group").
+
+A "project group" is a group of people working on a particular project at the
+behest of an official Rust team. Project groups must have:
+
+* A **charter** defining the project's scope
+* A **liaison** with an official Rust team (who may or may not also be a shepherd)
+* A small number of **shepherds**, who are responsible for summarizing
+  conversations and keeping the lang team abreast of interesting developments.
+* A GitHub repository hosted under the `rust-lang` organization containing the
+  charter and instructions for how community members can monitor the group's
+  progress and/or participate.
+
+[This blog post][shepherds-3.0] explains in detail the role of the
+shepherds.
+
+## Project group roadmap and RFCs
+
+The first step of the project group is to define a **roadmap** indicating the
+planned sequence in which it will design and propose particular behaviors and
+features.  Once the project group feels it has completed work on some item in
+the roadmap, that item will be submitted to the community as an RFC or FCP.
```

Comment:
```suggestion
the roadmap, that item will be submitted to the community as an RFC or FCP,
and to be reviewed by the lang team.
```

Reply:
I was considering the lang team to be part of "the community" here. Does this need to be called out explicitly, since, ultimately, the lang team needs to weigh in on such RFCs anyway? (Is that correct?)

Reply:
I only wanted to add this because the current wording suggests that the PG would be the ones approving of the FCP. Here's an alternate wording:
```suggestion
the roadmap, that item will be submitted as an RFC or FCP for review by the lang team and the community at large.
```